portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,21 +14,42 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns messages/ comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private List<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+    messages.add("Green tea was founded in China in 2737 B.C.");
+    messages.add("English Breakfast Tea came into existence in 1843");
+    messages.add("Chai tea was invented between 5000 and 9000 years ago");
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String titleText = "<h1>Welcome to teatime @ trin's!</h1>";
+    String json = convertToJson(messages);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
 
-    response.setContentType("text/html;");
-    response.getWriter().println(titleText);
+  /**
+   * Converts the messages list instance into a JSON string using the Gson library. 
+   */
+  private String convertToJson(List<String> messages) {
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
```

Comment:
There is no need to declare a variable for this string, you can just return it right away.

Reply:
"return new Gson().toJson(messages);" if you really want to make it concise, but "return gson.toJson(messages);" also works

Reply:
I've added that now so it'll be concise in my next commit! :) Thank you!!